packages/analytics-js/src/components/core/Analytics.ts

```diff
@@ -401,18 +412,18 @@ class Analytics implements IAnalytics {
     const type = 'ready';
     this.errorHandler.leaveBreadcrumb(`New ${type} invocation`);
 
-    if (!isFunction(callback)) {
-      this.logger.error(READY_API_CALLBACK_ERROR(READY_API));
+    if (!state.lifecycle.loaded.value) {
+      state.eventBuffer.toBeProcessedArray.value.push([type, callback]);
       return;
     }
 
-    if (!state.lifecycle.loaded.value) {
-      state.eventBuffer.toBeProcessedArray.value.push([type, callback]);
+    if (!isFunction(callback)) {
```

Comment:
why we changed order here? we would not want to process the callback if it is not a function